projects/pedal/src/throttle.c

```diff
@@ -0,0 +1,59 @@
+#include "throttle.h" 
+
+#include <string.h> 
+#include <stdio.h> 
+#include <stdbool.h> 
+#include <limits.h>
+
+#include "event_queue.h"
+#include "log.h"
+#include "pedal_events.h" 
+#include "soft_timer.h"
+
+
+
+static void prv_timer_callback(SoftTimerId timer_id, void *context){
+    ThrottleStorage *storage = context; 
+    int16_t reading = INT16_MAX; 
+    
+    if(storage->state == THROTTLE_STATE_ENABLE){
```

Comment:
Disable the timer when the throttle is off, instead of checking every single time for the throttle flag. This results in a whole lot of unnecessary calculations every 10ms which isn't good.